packages/react-aria-components/src/Tree.tsx

```diff
@@ -0,0 +1,548 @@
+// need to do a bunch of check to figure out what is the Content and what are the actual collection elements (aka child rows) of the TreeItem
+export interface TreeItemContentProps extends Pick<RenderProps<TreeItemContentRenderProps>, 'children'> {}
+
+// TODO does this need ref or context? Its only used to shallowly render the Content node... If it was a more generic collection component then I could see an argument for it
+// having those
+function TreeItemContent(props: TreeItemContentProps) {
+  let ref = useRef(null);
+  let shallow = useShallowRender('content', props, ref);
+  if (shallow) {
+    return shallow;
+  }
+}
+
+// TODO: is this pure needed?
+const _TreeItemContent = /*#__PURE__*/ TreeItemContent;
+export {_TreeItemContent as TreeItemContent};
+
+function TreeRow<T>({item}: {item: Node<T>}) {
+  let state = useContext(TreeStateContext)!;
+  let ref = useObjectRef<HTMLDivElement>(item.props.ref);
+  let {rowProps, gridCellProps, ...states} = useTreeGridListItem({node: item}, state, ref);
+  let stringFormatter = useLocalizedStringFormatter(intlMessages, 'react-aria-components');
+  let isExpanded = rowProps['aria-expanded'] === true;
+  let hasChildRows = [...state.collection.getChildren!(item.key)]?.length > 1;
+  let level = rowProps['aria-level'];
+
+  let {hoverProps, isHovered} = useHover({
+    isDisabled: !states.allowsSelection && !states.hasAction
+  });
+
+  let {isFocusVisible, focusProps} = useFocusRing();
+  let {checkboxProps} = useGridListSelectionCheckbox(
+    {key: item.key},
+    state
+  );
+
+  let props: TreeItemProps<unknown> = item.props;
+  let renderPropValues = React.useMemo(() => ({
+    ...states,
+    isHovered,
+    isFocusVisible,
+    isExpanded,
+    hasChildRows,
+    level,
+    selectionMode: state.selectionManager.selectionMode,
+    selectionBehavior: state.selectionManager.selectionBehavior
+  }), [states, isHovered, isFocusVisible, state.selectionManager, isExpanded, hasChildRows, level]);
+
+  let renderProps = useRenderProps({
+    ...props,
+    id: undefined,
+    children: item.rendered,
+    defaultClassName: 'react-aria-TreeItem',
+    values: renderPropValues
+  });
+
+  useEffect(() => {
+    if (!item.textValue) {
+      console.warn('A `textValue` prop is required for <TreeItem> elements in order to support accessibility features such as type to select.');
+    }
+  }, [item.textValue]);
+
+
+  let expandButtonProps = {
+    onPress: () => {
+      if (!states.isDisabled) {
+        state.toggleKey(item.key);
+      }
+    },
+    'aria-label': isExpanded ? stringFormatter.format('collapse') : stringFormatter.format('expand'),
+    // TODO: the below actually isn't enough to have keyboard navigation skip over it, we need it to be a span type button but
+    // RAC buttons are always a "button" type element.
+    tabIndex: isAndroid() ? -1 : null
```

Comment:
will need to figure out a way to accomplish this. Tableview is able to skip over its chevron button because it is a span type button and it overrides the tabIndex from the button hook